commit two

PupZemli-code opened the pull request:
написал основные функции, работает на внутреннем тесте

PupZemli-code committed:
первый термин, программа работает на тесте

dirtymew commented on internal/server/server.go:
```diff
@@ -1 +1,54 @@
+	}
+}
+
+func InitLogger() *log.Logger {
```
это не совсем к серверу относится....

dirtymew commented on internal/service/service.go:
```diff
@@ -1 +1,30 @@
+		return "", errors.New("файл пуст")
+	}
+
+	if strings.Contains(string(msg), "..") ||
```
лучше ContainsFunc

dirtymew commented on internal/service/service.go:
```diff
@@ -1 +1,30 @@
+	}
+
+	if strings.Contains(string(msg), "..") ||
+		strings.Contains(string(msg), "--") ||
```
это два тире

алфавит морзе это  только `.` `-`

PupZemli-code replied:
я понял что морзе всегда начинается с . или -

dirtymew replied:
ну не только начинается а содержит только их

dirtymew commented on internal/handlers/handlers.go:
```diff
@@ -1 +1,107 @@
+	}
+
+	/*// 6. Генерация имени файла
+	timestamp := time.Now().UTC().String()
```
почему закоментировано?

это нужно по условию задачи

dirtymew commented on internal/handlers/handlers.go:
```diff
@@ -1 +1,107 @@
+	// 7. Создание и запись в локальный файл
+	//err = writeToFile(newFilename, convertedString)
+
+	fileRes, err := os.OpenFile("fileRes", os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0666)
```
файлы должны быть разные

dirtymew replied:
можно проще os.Create(time.Now().UTC().String() + filepath.Ext(fileHeader.Filename))

dirtymew commented on internal/handlers/handlers.go:
```diff
@@ -1 +1,107 @@
+
+	// 8. Возврат результата
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprintf(w, "Файл успешно конвертирован и сохранен как %s\n", "fileRes")
```
логируй только ошибки

dirtymew commented on internal/handlers/handlers.go:
```diff
@@ -1 +1,107 @@
+	// 8. Возврат результата
+	w.WriteHeader(http.StatusOK)
+	fmt.Fprintf(w, "Файл успешно конвертирован и сохранен как %s\n", "fileRes")
+	fmt.Fprintf(w, "Результат: %s", convertedString)
```
не обрабатываешь ошибку

dirtymew replied:
ошибка так и не обработана @PupZemli-code

PupZemli-code replied:
Потерялся мой комментарий. Я не понял какую ошибку тут нужно обрабатывать, проверить метод заголовка?

dirtymew replied:
https://pkg.go.dev/fmt#Fprintf 

возвращает ошибку @PupZemli-code

PupZemli-code replied:
Понял, спасибо

dirtymew commented on internal/handlers/handlers.go:
```diff
@@ -1 +1,107 @@
+func writeToFile(filename string, content string) error {
+	// Создание директории, если она не существует
+	dir := filepath.Dir(filename)
+	if dir != "go1fl-sprint6-tpl/Data_fail" {
```
это что-то странное - пиши просто в текущую рабочую директорию

dirtymew commented on internal/handlers/handlers.go:
```diff
@@ -1 +1,107 @@
+	}
+
+	// 3. Получение файла из формы
+	file, _, err := r.FormFile("myFile")
```
file, fileHeader , err

dirtymew reviewed (changes requested):
Привет Михаил

по работе есть замечания - нужно поправить

PupZemli-code committed:
commit 2
Переделал пакет service через startings.ContainsFunc, а так же перенес initLogger из servir в main

PupZemli-code committed:
commit 3
Переработал handlers убрал лишние функции, исправил недочеты

PupZemli-code committed:
commit 4
Настроил логирование

dirtymew commented on cmd/main.go:
```diff
@@ -1,5 +1,45 @@
+	if err != nil {
+		logger.Fatal("Ошибка запуска сервера:", err)
+	}
+	logger.Println("Сервер запущен")
```
это нужно на 39 строку иначе никогда печататься не будет

dirtymew commented on internal/handlers/handlers.go:
```diff
@@ -1 +1,109 @@
+		logger.Println("Метод не поддерживается")
+		return
+	}
+	logger.Println("Метод проверен успешно")
```
не нужно так делать - логируй только ошибки

PupZemli-code replied:
Я думал это уровень INFO его ведь тоже логируют?

dirtymew replied:
логируют, но нужен логгер с поддержкой уровней логирования - лишние логи не помогают а обычно мешают.

правильная организация логирования это отдельная большая тема - к сожалению за рамками базового курса @PupZemli-code

PupZemli-code replied:
Круто. Спасибо, принято

dirtymew reviewed (approved):
Михаил

работа принята
